Sources/muterCore/TestReporting/Logger.swift

```diff
@@ -0,0 +1,147 @@
+    }
+    
+    func projectCoverageDiscoveryStarted() {
+        print("Running tests with coverage enabled to determine which files to mutate")
+    }
+    
+    func projectCoverageDiscoveryFinished(success: Bool) {
+        guard success == false else { return }
+        
+        print(
+                """
+                Gathering coverage failed.
+                Proceeding with mutation testing anyway.
+                Pass --skip-coverage argument to disable this step
+                """
+        )
+    }
+    
+    func sourceFileDiscoveryStarted() {
+        printMessage("Discovering Swift files which Muter will analyze...")
+    }
+    
+    func sourceFileDiscoveryFinished(sourceFileCandidates: [String]) {
+        let fileNames = sourceFileCandidates
+            .map(URL.init(fileURLWithPath:))
+            .map { $0.lastPathComponent }.joined(separator: "\n").bold
+        
+        print("In total, Muter discovered \(sourceFileCandidates.count) Swift files\n\n\(fileNames)")
+    }
+    
+    func mutationPointDiscoveryStarted() {
+        printMessage("Analyzing source files to find mutants which can be inserted into your project...")
+    }
+    
+    func mutationPointDiscoveryFinished(mutationPoints: [MutationPoint]) {
+        numberOfMutationPoints = mutationPoints.count
+        let numberOfFiles = mutationPoints.map { $0.fileName }.deduplicated().count
+        
+        print("In total, Muter discovered \(mutationPoints.count) mutants in \(numberOfFiles) files\n")
+        for (fileName, mutantCount) in mutationPointsByFileName(from: mutationPoints) {
+            print("\(fileName) (\(mutantCount) mutants)".bold)
+        }
+    }
+    
+    private func mutationPointsByFileName(from mutationPoints: [MutationPoint]) -> [String: Int] {
+        var result: [String: Int] = [:]
+        
+        for mutationPoint in mutationPoints {
+            if result[mutationPoint.fileName] == nil {
+                result[mutationPoint.fileName] = 1
+                continue
+            }
+            
+            result[mutationPoint.fileName]! += 1
+        }
+        return result
+    }
+    
+    func mutationTestingStarted() {
+        printMessage("Mutation testing will now begin\nRunning your test suite to determine a baseline for mutation testing...")
+    }
+    
+    func newMutationTestLogAvailable(mutationTestLog: MutationTestLog) {
+        if mutationTestLog.mutationPoint == nil {
+            print("""
+                Determined baseline for mutation testing
+                Muter is now going to apply each mutant one at a time and run your test suite for each mutant
+                After this step, Muter will generate a report detailing the efficacy of your test suite
+                This step may take a while
+                
+                
+                """)
+            progressBar = ProgressBar(
```

Comment:
I don't know what this ProgressBar thing is, but it's just screaming to be refactored into a result builder 😆